src/custom/components/CowBalanceButton/index.tsx

```diff
@@ -80,25 +79,23 @@ interface CowBalanceButtonProps {
   isUpToSmall?: boolean
 }
 
-const COW_DECIMALS = COW[ChainId.MAINNET].decimals
-
 export default function CowBalanceButton({ onClick, isUpToSmall }: CowBalanceButtonProps) {
   const { chainId } = useWeb3React()
   const { balance, isLoading } = useCombinedBalance()
 
-  const formattedBalance = formatSmartLocaleAware(balance, 0)
-  const formattedMaxBalance = formatMax(balance, COW_DECIMALS)
-
   if (!supportedChainId(chainId)) {
     return null
   }
 
+  // We don't need to display decimals symbols, so, we round the balance amount
+  const roundedAmount = FractionUtils.round(balance)
+
   return (
     <Wrapper isLoading={isLoading} onClick={onClick}>
       <CowProtocolLogo />
       {!isUpToSmall && (
-        <b title={formattedMaxBalance && `${formattedMaxBalance} (v)COW`}>
-          <Trans>{formattedBalance || 0}</Trans>
+        <b>
+          <TokenAmount amount={roundedAmount} defaultValue="0" tokenSymbol={balance?.currency} />
```

Comment:
Here we might want to keep the `(v)COW`, as it's the sum of vCOW and COW balances

Reply:
Nice catch! Thanks!